3-networks/modules/transitivity/main.tf

```diff
@@ -0,0 +1,181 @@
+}
+
+module "migs" {
+  source            = "terraform-google-modules/vm/google//modules/mig"
+  version           = "6.0.0"
+  for_each          = toset(var.regions)
+  project_id        = var.project_id
+  region            = each.key
+  target_size       = 3
+  hostname          = "transitivity-gw-${each.key}"
+  instance_template = module.instance_templates[each.key].self_link
+  update_policy = [
+    {
+      max_surge_fixed              = 3
+      max_surge_percent            = null
+      instance_redistribution_type = "NONE"
+      max_unavailable_fixed        = 3
+      max_unavailable_percent      = null
+      min_ready_sec                = 180
+      minimal_action               = "RESTART"
+      type                         = "OPPORTUNISTIC"
+    }
+  ]
+}
+
+module "firewall" {
+  source     = "github.com/terraform-google-modules/cloud-foundation-fabric//modules/net-vpc-firewall?ref=v4.3.0"
+  project_id = var.project_id
+  network    = var.vpc_name
+  # we do not use tag based firewall rules
+  ssh_source_ranges   = []
+  http_source_ranges  = []
+  https_source_ranges = []
+  # we use custom rules instead
+  custom_rules = {
+    allow-ssh-hcs = {
+      description = "Allow SSH to transitivity gateways"
+      direction   = "INGRESS"
+      action      = "allow"
+      sources     = []
+      ranges = [
+        # Health checkers ranges
+        "35.191.0.0/16", "130.211.0.0/22", "209.85.152.0/22", "209.85.204.0/22"
+      ]
+      targets              = [module.service_account.emails["transitivity-gw"]]
+      use_service_accounts = true
+      rules                = [{ protocol = "tcp", ports = [22] }]
+      extra_attributes     = {}
+    },
+    # Note: Actual allowed traffic is controlled via iptables within the guest environment.
+    allow-all-internal-ingress = {
+      description          = "Allow all private traffic to transitivity gateways"
+      direction            = "INGRESS"
+      action               = "allow"
+      sources              = []
+      ranges               = flatten(values(var.regional_aggregates))
+      targets              = [module.service_account.emails["transitivity-gw"]]
+      use_service_accounts = true
+      rules                = [{ protocol = "all", ports = null }]
+      extra_attributes     = {}
+    }
+    allow-all-internal-egress = {
+      description          = "Allow all private traffic to transitivity gateways"
+      direction            = "EGRESS"
+      action               = "allow"
+      sources              = []
+      ranges               = flatten(values(var.regional_aggregates))
+      targets              = [module.service_account.emails["transitivity-gw"]]
+      use_service_accounts = true
+      rules                = [{ protocol = "all", ports = null }]
+      extra_attributes     = {}
+    }
+  }
+}
+
+module "ilb_addresses" {
+  source     = "github.com/terraform-google-modules/cloud-foundation-fabric//modules/net-address?ref=v4.3.0"
```

Comment:
We should not reference Fabric modules from a different repo. They're not a stable interface.